src/resource.h

```diff
@@ -79,6 +81,16 @@ class Resource {
   /// @return a new resource object with same state id and quantity == quantity
   virtual Ptr ExtractRes(double quantity) = 0;
 
+  /// To enable the Decay method to be called on any child resource, define
+  /// a null op Decay method here.
+  /// @param curr_time the current time for the decay oepration
+  virtual void Decay(int curr_time) { throw Error("cannot decay resource type " + this->type()); };
+
+  /// To enable the Absorb method to be called on any child resource, define
+  /// a null op Absorb method here.
```

Comment:
`op`? Is that a typo in the documentation?

Reply:
@gonuke this is from #1687

Reply:
`null op` is a common term to describe a function that does nothing.  That said, I wrote this comment when the function really did do nothing (it just returned) so this comment should probably be udpated.